common/persistence/sql/sql_task_store.go

```diff
@@ -189,31 +189,92 @@ func (m *sqlTaskStore) LeaseTaskList(
 		if rowsAffected == 0 {
 			return fmt.Errorf("%v rows affected instead of 1", rowsAffected)
 		}
+		var c *persistence.TaskListPartitionConfig
+		if tlInfo.AdaptivePartitionConfig != nil {
+			c = &persistence.TaskListPartitionConfig{
+				Version:            tlInfo.AdaptivePartitionConfig.Version,
+				NumReadPartitions:  int(tlInfo.AdaptivePartitionConfig.NumReadPartitions),
+				NumWritePartitions: int(tlInfo.AdaptivePartitionConfig.NumWritePartitions),
+			}
+		}
 		resp = &persistence.LeaseTaskListResponse{TaskListInfo: &persistence.TaskListInfo{
-			DomainID:    request.DomainID,
-			Name:        request.TaskList,
-			TaskType:    request.TaskType,
-			RangeID:     rangeID + 1,
-			AckLevel:    ackLevel,
-			Kind:        request.TaskListKind,
-			LastUpdated: now,
+			DomainID:                request.DomainID,
+			Name:                    request.TaskList,
+			TaskType:                request.TaskType,
+			RangeID:                 rangeID + 1,
+			AckLevel:                ackLevel,
+			Kind:                    request.TaskListKind,
+			LastUpdated:             now,
+			AdaptivePartitionConfig: c,
 		}}
 		return nil
 	})
 	return resp, err
 }
 
+func (m *sqlTaskStore) GetTaskList(
+	ctx context.Context,
+	request *persistence.GetTaskListRequest,
+) (*persistence.GetTaskListResponse, error) {
+	dbShardID := sqlplugin.GetDBShardIDFromDomainIDAndTasklist(request.DomainID, request.TaskList, m.db.GetTotalNumDBShards())
+
+	domainID := serialization.MustParseUUID(request.DomainID)
```

Comment:
not sure about using MustParse here, this implies malformed data will panic. What about just doing a UUID parse and returng an error if it is invalid? I'd hate to see a daemon for handling the tasklist scaling panic due to some bad data and crashing the entire matching service.

Reply:
this is used everywhere, just want to keep everything consistent

Reply:
It seems to only used within sql package. Not ideal but let's keep it consistent for now. It's unlikely to hit panic in those places unless someone manually writes to db